x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java

```diff
@@ -293,19 +295,18 @@ public void getRoleMappings(Set<String> names, ActionListener<List<ExpressionRol
     }
 
     private void getMappings(ActionListener<List<ExpressionRoleMapping>> listener) {
-        if (securityIndex.isAvailable()) {
-            loadMappings(listener);
-        } else {
-            logger.info("The security index is not yet available - no role mappings can be loaded");
-            if (logger.isDebugEnabled()) {
-                logger.debug(
-                    "Security Index [{}] [exists: {}] [available: {}]",
-                    SECURITY_MAIN_ALIAS,
-                    securityIndex.indexExists(),
-                    securityIndex.isAvailable()
-                );
-            }
+        final SecurityIndexManager frozenSecurityIndex = securityIndex.defensiveCopy();
```

Comment:
this is one the motivation for this change. in stateless, during startup, this would appear with some regularity during the startup. it is believed that this because the old check only looked at the primary shards we only care about the search shards and in stateless there is a meaningful difference since primaries/index can not be searched resulting in this and downstream errors in the log. It worth noting this is called for every realm that needs role mapping during authentication so this is called with high frequency.